mobile/src/main/java/org/openhab/habdroid/util/HttpClient.kt

```diff
@@ -54,10 +54,11 @@ class HttpClient constructor(client: OkHttpClient, baseUrl: String?, username: S
         if (authHeader != null) {
             // Forcibly put authorization info into request, as redirect/retry interceptor might have removed it
             clientBuilder.addNetworkInterceptor { chain ->
-                val request = chain.request()
-                    .newBuilder()
-                    .addHeader("Authorization", authHeader)
-                    .build()
+                var request = chain.request()
+                // Don't add auth header to requests to external icons
+                if (baseUrl != null && request.url.toString().startsWith(baseUrl)) {
+                    request = request.newBuilder().addHeader("Authorization", authHeader).build()
+                }
```

Comment:
I find working with immutable values preferable. Thus my suggestion would be something like
```
            clientBuilder.addNetworkInterceptor { chain ->
                val request = chain.request()
                // Make sure not to send authentication credentials to external servers
                chain.proceed(if (this.baseUrl?.host == request.url.host) {
                    request.newBuilder().addHeader("Authorization", authHeader).build()
                } else {
                    request
                })
            }
```

Reply:
Nice! Thanks for the suggestion